components/scream/src/share/atm_process/atmosphere_process.hpp

```diff
@@ -149,11 +159,14 @@ class AtmosphereProcess : public ekat::enable_shared_from_this<AtmosphereProcess
   const std::list<const_group_type>& get_groups_in  () const { return m_groups_in;  }
   const std::list<      group_type>& get_groups_out () const { return m_groups_out; }
 
-  // Whether this atm proc requested the field/group as in/out/inout, via a FieldRequest/GroupRequest.
-  bool requires_field (const FieldIdentifier& id) const;
-  bool computes_field (const FieldIdentifier& id) const;
-  bool requires_group (const std::string& name, const std::string& grid) const;
-  bool computes_group (const std::string& name, const std::string& grid) const;
+  // The base class does not store internal fields.
+  virtual const std::list<field_type>& get_internal_fields  () const { return m_internal_fields; }
+
+  // Whether this atm proc requested the field/group as in/out, via a FieldRequest/GroupRequest.
+  bool is_required_field (const FieldIdentifier& id) const;
```

Comment:
Not a big deal at all, but is there a reason you decided to change these method names? To me, `atm_proc.requires_field(f)` is a nice example of  the `noun.relates_to(other_noun)` idiom.

Reply:
I didn't have an equivalent for "internal" fields... So to keep the impl for required, computed, internal somewhat parallel, I switched to this wording. I can change to `has_blah_field(f)`, with blah in `required, computed, internal` maybe?

Reply:
Aha, makes sense. I like `atm_proc.has_blah_field(f)`, actually. If that's okay with you, let's do it!